### DISCUSSION TITLE
Static website deployment on gitlab pages

### DISCUSSION THREAD

**Discussion by vineetpandey** [awslabs/threat-composer#43] [General]
Rather than publishing on AWS, would like to try in gitlab pages. My query is whether deployment on gitlab pages is seamless and would work with all the features threat composer have? @jessieweiyi

**Comment by dboyd13** [comment]
As of today ([Release v1.0.18](https://github.com/awslabs/threat-composer/releases/tag/v1.0.18)) the deployed Threat Composer web application is identical for the [GitHub Pages version](https://awslabs.github.io/threat-composer/) and if one were to deploy using the included [CDK project](https://github.com/awslabs/threat-composer#deployment) to AWS. Whilst we have not tested this using [GitLab Pages](https://docs.gitlab.com/ee/user/project/pages/), I do not see why it would not work and you may be able to reuse portions of our [GitHub pages build and deploy workflow definitions](https://github.com/awslabs/threat-composer/tree/main/.github/workflows) to get started quicker.

Note the [Security Considerations](https://github.com/awslabs/threat-composer#security-considerations) section of the README, in particular around scoping network accessibility, authentication and HSTS.

**Reply by vineetpandey** [reply]
Thanks for the detailed information.
I went through the security requirements already. Idea here is to deploy using gitlab pages and to validate in terms of functionality and to ensure no difference in features for gitlab pages and AWS deployment(except not meeting security requirements). If everything looks good then will move to AWS.

**Reply by dboyd13** [reply]
Sure. I may be missing some context of your use-case, but you could consider evaluating the tool using the existing Git**Hub** pages hosted version https://awslabs.github.io/threat-composer - then deploy to AWS if it meets your needs.

**Reply by vineetpandey** [reply]
Thanks. With some significant effort, I'm able to deploy this tool on gitlab pages - [threat-composer](https://vineetpandey.gitlab.io/threat-composer/workspaces/default/dashboard). 
Will explore the tool in detail.

Also let me know about any possibility if I can contribute with my gitlab CI/CD code.

**Reply by dboyd13** [reply]
Awesome, glad to hear you managed to get it done. As for the GitLab CI/CD definitions perhaps you want to put into into a [GitHub Gist](https://gist.github.com/) and link to it from this discussion thread?

**Reply by vineetpandey** [reply]
Hi @dboyd13 - Apologies, I totally missed creating/linking the gist. While with recent releases using PDK, I had to recreate the .gitlab-ci.yml file. One can check out this Gist for more details: **[ .gitlab-ci.yml](https://gist.github.com/vineetpandey/3aacadf9517daabcfbfd81fb83f1b1be)**
